src/range.rs

```diff
@@ -2,12 +2,42 @@
 //! (`Range<T>`); this is the type for values created by a `a..b`
 //! expression. You will rarely need to interact with it directly
 //! unless you have need to name one of the iterator types.
+//! 
+//! ```
+//! use rayon::prelude::*;
+//! 
+//! let r = (0..100u64).into_par_iter()
+//!                    .fold(|| 0, |acc, cur| acc + cur)
+//!                    .sum();
+//! 
+//! // compare result with sequential calculation
+//! assert_eq!((0..100).sum::<u64>(), r);
+//! ```
 
 use iter::*;
 use iter::internal::*;
 use std::ops::Range;
 
 /// Parallel iterator over a range
+/// Implemented for `u32`, `i32`, `usize` and `isize`.
```

Comment:
We should have a blank doc-line between the intro and additional description.  Or if you want it as one intro paragraph, then it needs punctuation between the clauses.

Also, `ParallelIterator` is implemented for `range::Iter<T>` of _all_ integer types.  The difference is just that `i64` and `u64` don't implement `IndexedParallelIterator`, which you'd need for `zip`, for example.